libterminfo/curterm.c

```diff
@@ -99,21 +99,21 @@ set_curterm(TERMINAL *nterm)
 
 		p = ttytype;
 		l = sizeof(ttytype);
-		if ((n = snprintf(p, l, "%s", nterm->name)) == strlen(p)) {
+		if ((n = snprintf(p, l, "%s", nterm->name)) < l) {
```

Comment:
good change, but can we be certain tht snprintf won't return -1 due to an error ?

Reply:
I think the only possible error is EOVERFLOW if `strlen(nterm->name) > INT_MAX`. But I think terminfo string lengths are limited to 16-bit, so this shouldn't be possible.

An alternative is to use memccpy. Something like this:

```diff
diff --git a/libterminfo/curterm.c b/libterminfo/curterm.c
index cc42fbf..9fc51fa 100644
--- a/libterminfo/curterm.c
+++ b/libterminfo/curterm.c
@@ -80,8 +80,7 @@ set_curterm(TERMINAL *nterm)
 {
 	TERMINAL *oterm;
 	size_t l;
-	int n;
-	char *p;
+	char *p, *q;
 
 	oterm = cur_term;
 	cur_term = nterm;
@@ -99,26 +98,23 @@ set_curterm(TERMINAL *nterm)
 
 		p = ttytype;
 		l = sizeof(ttytype);
-		if ((n = snprintf(p, l, "%s", nterm->name)) == strlen(p)) {
-			p += n;
-			l -= n;
-			*p++ = '|';
-			l--;
+		if ((q = memccpy(p, nterm->name, '\0', l))) {
+			l -= q - p;
+			p = q;
+			p[-1] = '|';
 			if (nterm->_alias  &&
-				(n = snprintf(p, l, "%s", nterm->_alias)) == strlen(p))
+				(q = memccpy(p, nterm->_alias, '\0', l)))
 			{
-				p += n;
-				l -= n;
-				*p++ = '|';
-				l--;
+				l -= q - p;
+				p = q;
+				p[-1] = '|';
 			}
 			if (nterm->desc  &&
-				(n = snprintf(p, l, "%s", nterm->desc)) == strlen(p))
+				(q = memccpy(p, nterm->desc, '\0', l)))
 			{
-				p += n;
-				l -= n;
-				*p++ = '|';
-				l--;
+				l -= q - p;
+				p = q;
+				p[-1] = '|';
 			}
 			p--;
 		}
```